### PR TITLE
Localbenchmarks

### PR DESCRIPTION
**Pull request by tyurek**
adds a script that can do local benchmarking for avss. Allows for precomputation with polycommits. Also adds earlier termination for rbc with client mode

### CODE REVIEW

File: tests/test_hbavss.py [hunk 1]
```diff
@@ -603,6 +647,12 @@ def _get_dealer_msg(self, values, n):
 
 @mark.asyncio
 async def test_hbavss_light_client_mode(test_router):
+    def callback(future):
```
Comment (amiller) [inline tests/test_hbavss.py]: We can now import this as:
`from honeybadgermpc.utils.misc import print_exception_callback`

File: pairing/src/lib.rs [hunk 2]
```diff
@@ -611,13 +611,15 @@ impl PyFr {
     //    })
     //}
     //fn __new__(obj: &PyRawObject, s1: u32, s2: u32, s3: u32, s4: u32) -> PyResult<()>{
-    fn __new__(obj: &PyRawObject, s: &str) -> PyResult<()>{
-        let f =  Fr::from_str(s).unwrap();
```
Comment (amiller) [inline pairing/src/lib.rs:615]: let's get rid of the commented out code

File: benchmark/test_benchmark_hbavss.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+    (t, n, g, h, pks, sks, crs, pc, values, field) = params
+    sends, recvs, _ = test_router(n+1)
+    dealer_id = n
+
+    hbavss = HbAvssLight(pks, None, crs, n, t, dealer_id, sends[dealer_id], recvs[dealer_id], pc=pc, field=field)  # (# noqa: E501)
+    await asyncio.create_task(hbavss.avss(0, value=values, client_mode=True))
+
+
+async def hbavss_batch_batch(test_router, params):
```
Comment (amiller) [inline benchmark/test_benchmark_hbavss.py]: Can't help but think `batch_batch` sounds redundant. Let's consider an alternate name for a future commit

**Review by amiller — CHANGES_REQUESTED** [review]
This should ideally be two commits. One is about adding benchmarks, the other is a bunch of changes/improvements to the AVSS implementation.

File: honeybadgermpc/betterpairing.py [hunk 4]
```diff
@@ -111,7 +111,7 @@ def __pow__(self, other):
             try:
                 intother = int(other)
                 exponend = ZR(intother)
-            except:
+            except Exception:
```
Comment (amiller) [inline honeybadgermpc/betterpairing.py]: This is too broad - let's figure out what a type error would throw and just catch that

File: tests/test_hbavss.py (hunk 1 above)
Comment (tyurek) [reply tests/test_hbavss.py]: oh cool. that's a good change

File: benchmark/test_benchmark_hbavss.py (hunk 3 above)
Comment (tyurek) [reply benchmark/test_benchmark_hbavss.py]: going with multibatch for now

**Review by amiller — APPROVED** [review]
This looks ready to go. Approve +1

**Comment by Drake-Eidukas** [comment]
Luckily there's no merge conflicts. I'll update travis.

**Pull request merged**